src/evo/deterministicmns.cpp

```diff
@@ -1096,12 +1094,7 @@ CDeterministicMNList CDeterministicMNManager::GetListForBlockInternal(gsl::not_n
 
     for (const auto& diffIndex : listDiffIndexes) {
         const auto& diff = mnListDiffsCache.at(diffIndex->GetBlockHash());
-        if (diff.HasChanges()) {
-            snapshot = snapshot.ApplyDiff(diffIndex, diff);
-        } else {
-            snapshot.SetBlockHash(diffIndex->GetBlockHash());
-            snapshot.SetHeight(diffIndex->nHeight);
-        }
+        snapshot.ApplyDiff(diffIndex, diff);
```

Comment:
Am I correct that this is the only place we have a performance improvement right? All other instances, just now have an explicit copy instead of implicit copy. This change does seem worth it, but only this specific one in GetListForBlockInternal has any perf improvement

Reply:
not counting unit tests, there are only 2 usages of `ApplyDiff`: `GetListForBlockInternal` and in `UndoBlock`. GetListForBlockInternal will be benefitial from this optimisation; undo block - doesn't become slower either faster.